### PR TITLE
InsightToolkit4: new port based on InsightToolkit-devel, pinned on v4

### PR DESCRIPTION
**Pull request by nilason**
#### Description

New `InsightToolkit4` port pinned on version 4.x with Python subports.

The latest version of [Orfeo ToolBox](https://ports.macports.org/port/orfeotoolbox/details/) is not compatible with v5, in order to update that Portfile this pinned port is needed.

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 12.6.6 21G646 arm64
Xcode 14.2 14C18

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried a full install with `sudo port -vst install`?
- [x] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Comment by nilason** [comment]
Finally I managed this to pass the CIs too (the failed one is a timeout).

@mascguy or @reneeotten, perhaps one of you could look at this? I need this to be able to update OrfeoToolbox.

File: graphics/InsightToolkit4/Portfile [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+long_description    {*}${description} is an open-source, cross-platform system that \
+                    provides developers with an extensive suite of software tools \
+                    for image analysis. Developed through extreme programming \
+                    methodologies, ITK builds on a proven, spatially-oriented \
+                    architecture for processing, segmentation, and registration \
+                    of scientific images in two, three, or more dimensions.
+
+homepage            https://itk.org
+
+distname            InsightToolkit-${version}
```
Comment (mascguy) [inline graphics/InsightToolkit4/Portfile:29]: Particularly when you have devel and non-devel ports, the distfiles should be mirrored in the same location. This ensures there's no duplication (and wasted space) on various mirrors.

So you'll want to set `dist_subdir` to a common name across these ports, such as `dist_subdir InsightToolkit`.

Comment (mascguy) [reply graphics/InsightToolkit4/Portfile:29]: To illustrate what I mean, take a look at what's being mirrored for the various versions:

* https://distfiles.macports.org/InsightToolkit

* https://distfiles.macports.org/InsightToolkit312

* https://distfiles.macports.org/InsightToolkit314

There is significant duplication, which would be eliminated with use of `dist_subdir`:

BrainPart1.tgz 2018-05-04 03:36 5.6M
BrainPart1Rotated10Translated15.tgz 2018-05-04 03:36 4.9M
BrainPart1Rotated20Translated20.tgz 2018-05-04 03:36 4.3M
BrainPart2.tgz 2018-05-04 03:37 5.3M
BrainPart2Rotated10Translated15.tgz 2018-05-04 03:37 4.6M
BrainPart2Rotated20Translated20.tgz 2018-05-04 03:37 4.1M
BrainPart3.tgz 2018-05-04 03:37 5.8M
BrainPart3Rotated10Translated15.tgz 2018-05-04 03:37 5.1M
BrainPart3Rotated20Translated20.tgz 2018-05-04 03:37 4.5M
CableSwig-ITK-3.12.0.tar.gz 2015-08-29 19:52 18M
DoxygenInsightToolkit-3.12.0.tar.gz 2015-08-29 19:54 102M
InsightToolkit-3.12.0.tar.gz 2015-08-29 19:52 30M

Comment (nilason) [reply graphics/InsightToolkit4/Portfile:29]: Does this mean I should change setting `distname` to set `dist_subdir`? Or are both needed? I suspect the latter is the case, but before I push nonsense...

Comment (mascguy) [reply graphics/InsightToolkit4/Portfile:29]: > Does this mean I should change setting `distname` to set `dist_subdir`? Or are both needed? I suspect the latter is the case, but before I push nonsense... 

No, you still need `distname` too, and that doesn't change.

**Comment by nilason** [comment]
Now I reordered the PGs, removed Python 3.9, added `dist_subdir` and added myself as maintainer.
I believe it should be good to go.

**Review by mascguy — APPROVED** [review]

File: graphics/InsightToolkit4/Portfile (hunk 1 above)
Comment (nilason) [reply graphics/InsightToolkit4/Portfile:29]: > > Does this mean I should change setting `distname` to set `dist_subdir`? Or are both needed? I suspect the latter is the case, but before I push nonsense...
> 
> No, you still need `distname` too, and that doesn't change.

Thought so. Thanks, now I understand the purpose with `dist_subdir`.

**Comment by nilason** [comment]
I did a minor change, reflecting the latest change of #20399. This now `conflicts_build  InsightToolkit`.

**Comment by mascguy** [comment]
CI looks good now, merging.

**Comment by nilason** [comment]
Thanks Christopher!

**Pull request merged**